### PR TITLE
Initial Test Cases TDD.

### PR DESCRIPTION
**Pull request by Jawahar-A**
This PR covers the test cases for controller and service files.

### CODE REVIEW

File: Vehicle_Parking_System.sln
Comment (ChanduFrontline) [inline Vehicle_Parking_System.sln:1]: Check the naming convention for sln, projects and files and ensure they are as per the standards.

File: .gitignore
Comment (ChanduFrontline) [inline .gitignore:1]: add a proper git ignore for .net projects

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            objectResult.StatusCode.Should().Be(400);
+
+
+        }
+
+        [Fact]
+        public async Task ShoudlReturnOn_GetVehicleById_AndStatusAs200()
+        {
+            var mockVehicleService = new Mock<IVehicleService>();
+            var testVehicleId = "1";
```
Comment (ChanduFrontline) [inline Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: try to mock the id

File: Vehicle_Parking_System/Program.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+builder.Services.AddSingleton<IVehicleService, VehicleService>();
+builder.Services.AddControllers();
+builder.Services.AddAutoMapper(typeof(AutoMapping));
```
Comment (ChanduFrontline) [inline Vehicle_Parking_System/Program.cs]: Are we using automapper currently?

Comment (Jawahar-A) [reply Vehicle_Parking_System/Program.cs]: Yes, I am going use to map dto and model object, so I've added to mock the mapper in service class initially.

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs (hunk 1 above)
Comment (Jawahar-A) [reply Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: resolved

File: .gitignore
Comment (Jawahar-A) [reply .gitignore:1]: resolved

File: Vehicle_Parking_System.sln
Comment (Jawahar-A) [reply Vehicle_Parking_System.sln:1]: resolved

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+namespace Vehicle_Parking_Management.tests.Test.Controller
+{
+    public class VehicleControllerTest
+    {
+        private readonly Fixture _fixture = new Fixture();
+        [Fact]
+        public async Task ShouldReturnOn_GetAllVehicle_AndStatusAs200()
+        {
+            var mockVehicleService = new Mock<IVehicleService>();
+            var mockVehicle=_fixture.Create<List<VehicleModel>>();
```
Comment (rohitvipin) [inline Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: format your code

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                ReturnsAsync(mockVehicle);
+
+            var testController = new VehicleController(mockVehicleService.Object);
+            var result = await testController.GetAllVehicle();
+
+            mockVehicleService.Verify(service => service.getAllVehicles(), Times.Once());
+            result.Should().BeOfType<OkObjectResult>();
+            var objectResult = (OkObjectResult)result;
+            objectResult.Value.Should().BeOfType<List<VehicleModel>>();
+            objectResult.StatusCode.Should().Be(200);
```
Comment (rohitvipin) [inline Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: use named const from statuscodes

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            var mockVehicle=_fixture.Create<List<VehicleModel>>();
+            mockVehicleService.Setup(service => service.getAllVehicles()).
+                ReturnsAsync(mockVehicle);
+
+            var testController = new VehicleController(mockVehicleService.Object);
+            var result = await testController.GetAllVehicle();
+
+            mockVehicleService.Verify(service => service.getAllVehicles(), Times.Once());
+            result.Should().BeOfType<OkObjectResult>();
+            var objectResult = (OkObjectResult)result;
```
Comment (rohitvipin) [inline Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: use and - please go through docs of fluent assertions 
line 26 returns OkObjectResult

File: Vehicle_Parking_Management.tests/Test/Services/VehicleServiceTest.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Vehicle_Parking_Management.tests.Test.Services
+{
+    public  class VehicleServiceTest
+    {
+       
+        private  readonly Mock<IDatabaseContext> _mockDataBaseContext;
+        private  readonly Mock<IMapper> _mockMapper;
+        private readonly Fixture _fixture=new Fixture();
+    public VehicleServiceTest()
```
Comment (rohitvipin) [inline Vehicle_Parking_Management.tests/Test/Services/VehicleServiceTest.cs]: format

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs (hunk 3 above)
Comment (Jawahar-A) [reply Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: Resolved

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs (hunk 5 above)
Comment (Jawahar-A) [reply Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: resolved.

File: Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs (hunk 4 above)
Comment (Jawahar-A) [reply Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs]: resolved.

File: Vehicle_Parking_Management.tests/Test/Services/VehicleServiceTest.cs (hunk 6 above)
Comment (Jawahar-A) [reply Vehicle_Parking_Management.tests/Test/Services/VehicleServiceTest.cs]: resolved

File: Vehicle.Mangement.System/Controllers/VehicleController.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using VehicleManagementSystem.Models.Api;
+using VehicleManagementSystem.Services.Vehicle;
+
+namespace VehicleManagementSystem.Controllers
```
Comment (shwetab2023) [inline Vehicle.Mangement.System/Controllers/VehicleController.cs]: File scoped namespace

File: Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+            mockVehicleService.Verify(service => service.getAllVehicles(), Times.Once());
+
+            result.Should().BeOfType<OkObjectResult>().Which.Value.Should()
+                .BeOfType<List<VehicleModel>>().And.Subject.As<List<VehicleModel>>()
+                .Should().BeEquivalentTo(mockVehicle);
+
+            result.Should().BeOfType<OkObjectResult>().Which.StatusCode.Should().
+                Be(StatusCodes.Status200OK);
+
```
Comment (shwetab2023) [inline Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs:39]: Remove unnecessary space

File: Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+            mockVehicleService.Verify(service => service.getVehicle(testVehicleId), Times.Once());
+
+            result.Should().BeOfType<OkObjectResult>().Which.Value.Should().
+                Be(mockVehicle).And.Subject.As<VehicleModel>().Should().
+                BeEquivalentTo(mockVehicle);
+
+            result.Should().BeOfType<OkObjectResult>().Which.StatusCode.Should()
+                .Be(StatusCodes.Status200OK);
+        }
```
Comment (shwetab2023) [inline Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs:87]: Leave a line space between test cases

File: Vehicle.Management.System.Tests/Test/Services/VehicleServiceTest.cs [hunk 10]
```diff
@@ -0,0 +1,78 @@
+﻿using System.Linq.Expressions;
+using AutoFixture;
+using AutoMapper;
+using MongoDB.Driver;
+using Moq;
+using VehicleManagementSystem.Models.Api;
+using VehicleManagementSystem.Models.Data;
+using VehicleManagementSystem.Services.Database;
+using VehicleManagementSystem.Services.Vehicle;
+
```
Comment (shwetab2023) [inline Vehicle.Management.System.Tests/Test/Services/VehicleServiceTest.cs:11]: Format this file to remove unnecessary space and add space wherever necessary

File: Vehicle.Mangement.System/Controllers/VehicleController.cs [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+{
+    [Route("api")]
```
Comment (shwetab2023) [inline Vehicle.Mangement.System/Controllers/VehicleController.cs:9]: Remove extra spaces from here as well

File: Vehicle.Mangement.System/Models/Api/VehicleDto.cs [hunk 12]
```diff
@@ -0,0 +1,13 @@
+﻿namespace VehicleManagementSystem.Models.Api
+{
+    public class VehicleDto
```
Comment (shwetab2023) [inline Vehicle.Mangement.System/Models/Api/VehicleDto.cs]: Use record

File: Vehicle.Mangement.System/Models/Data/VehicleModel.cs [hunk 13]
```diff
@@ -0,0 +1,19 @@
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace VehicleManagementSystem.Models.Data
+{
```
Comment (shwetab2023) [inline Vehicle.Mangement.System/Models/Data/VehicleModel.cs:5]: Be consistant

File: Vehicle.Mangement.System/Services/Database/DatabaseContext.cs [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public class DatabaseContext(IOptions<VehicleDataBaseSettings> options, IMongoDatabase database) : IDatabaseContext
+    {
+        private readonly IMongoDatabase _database = database;
+        private readonly IOptions<VehicleDataBaseSettings> _options = options;
+        private IMongoCollection<VehicleModel>? _vehicles;
+
+        public IMongoCollection<VehicleModel> Vehicles =>
+            _vehicles ??= _database.GetCollection<VehicleModel>(_options.Value.CollectionName);
+
```
Comment (shwetab2023) [inline Vehicle.Mangement.System/Services/Database/DatabaseContext.cs]: Remove

File: Vehicle.Mangement.System/Controllers/VehicleController.cs (hunk 7 above)
Comment (Jawahar-A) [reply Vehicle.Mangement.System/Controllers/VehicleController.cs]: resolved

File: Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs (hunk 9 above)
Comment (Jawahar-A) [reply Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs:87]: resolved

File: Vehicle.Mangement.System/Controllers/VehicleController.cs (hunk 11 above)
Comment (Jawahar-A) [reply Vehicle.Mangement.System/Controllers/VehicleController.cs:9]: resolved

File: Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs (hunk 8 above)
Comment (Jawahar-A) [reply Vehicle.Management.System.Tests/Test/Controller/VehicleControllerTest.cs:39]: resolved

**Pull request merged**